# ros2_control uses this configuration for controller management
# This file is only for simulation
controller_manager:
  ros__parameters:

    update_rate: 100 # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    left_ur_arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    left_forward_velocity_controller:
      type: velocity_controllers/JointGroupVelocityController

    left_forward_position_controller:
      type: position_controllers/JointGroupPositionController

    left_gripper_controller:
      type: position_controllers/GripperActionController

    right_ur_arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    right_forward_velocity_controller:
      type: velocity_controllers/JointGroupVelocityController

    right_forward_position_controller:
      type: position_controllers/JointGroupPositionController

    right_gripper_controller:
      type: position_controllers/GripperActionController

# -----------------------------------------------------------------
left_ur_arm_controller:
  ros__parameters:
    joints:
      - left_shoulder_pan_joint
      - left_shoulder_lift_joint
      - left_elbow_joint
      - left_wrist_1_joint
      - left_wrist_2_joint
      - left_wrist_3_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.2
      goal_time: 0.0
      left_shoulder_pan_joint: { trajectory: 0.2, goal: 0.1 }
      left_shoulder_lift_joint: { trajectory: 0.2, goal: 0.1 }
      left_elbow_joint: { trajectory: 0.2, goal: 0.1 }
      left_wrist_1_joint: { trajectory: 0.2, goal: 0.1 }
      left_wrist_2_joint: { trajectory: 0.2, goal: 0.1 }
      left_wrist_3_joint: { trajectory: 0.2, goal: 0.1 }

left_forward_velocity_controller:
  ros__parameters:
    joints:
      - left_shoulder_pan_joint
      - left_shoulder_lift_joint
      - left_elbow_joint
      - left_wrist_1_joint
      - left_wrist_2_joint
      - left_wrist_3_joint
    interface_name: velocity

left_forward_position_controller:
  ros__parameters:
    joints:
      - left_shoulder_pan_joint
      - left_shoulder_lift_joint
      - left_elbow_joint
      - left_wrist_1_joint
      - left_wrist_2_joint
      - left_wrist_3_joint

left_gripper_controller:
  ros__parameters:
    joint: left_robotiq_85_left_knuckle_joint

# -----------------------------------------------------------------
right_ur_arm_controller:
  ros__parameters:
    joints:
      - right_shoulder_pan_joint
      - right_shoulder_lift_joint
      - right_elbow_joint
      - right_wrist_1_joint
      - right_wrist_2_joint
      - right_wrist_3_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.2
      goal_time: 0.0
      right_shoulder_pan_joint: { trajectory: 0.2, goal: 0.1 }
      right_shoulder_lift_joint: { trajectory: 0.2, goal: 0.1 }
      right_elbow_joint: { trajectory: 0.2, goal: 0.1 }
      right_wrist_1_joint: { trajectory: 0.2, goal: 0.1 }
      right_wrist_2_joint: { trajectory: 0.2, goal: 0.1 }
      right_wrist_3_joint: { trajectory: 0.2, goal: 0.1 }

right_forward_velocity_controller:
  ros__parameters:
    joints:
      - right_shoulder_pan_joint
      - right_shoulder_lift_joint
      - right_elbow_joint
      - right_wrist_1_joint
      - right_wrist_2_joint
      - right_wrist_3_joint
    interface_name: velocity

right_forward_position_controller:
  ros__parameters:
    joints:
      - right_shoulder_pan_joint
      - right_shoulder_lift_joint
      - right_elbow_joint
      - right_wrist_1_joint
      - right_wrist_2_joint
      - right_wrist_3_joint

right_gripper_controller:
  ros__parameters:
    joint: right_robotiq_85_left_knuckle_joint